python/ray/serve/air_integrations.py

```diff
@@ -57,19 +57,28 @@ def _load_predictor_cls(
     return predictor_cls
 
 
-def _unpack_tensorarray_from_pandas(output_df: "pd.DataFrame") -> "pd.DataFrame":
-    """Unpack predictor's return value with TensorArray into numpy.
+def _unpack_dataframe_to_serializable(output_df: "pd.DataFrame") -> "pd.DataFrame":
+    """Unpack predictor's pandas return value to JSON serializable format.
 
     In dl_predictor.py we return a pd.DataFrame that could have multiple
     columns but value of each column is a TensorArray. Flatten the
     TensorArray to list to ensure output is json serializable as http
     response.
+
+    In numpy predictor path, we might return collection of np.ndarrays that also
+    requires flattening to list to ensure output is json serializable as http
+    response.
     """
     from ray.data.extensions import TensorDtype
 
     for col in output_df.columns:
+        # TensorArray requires special handling to numpy array.
         if isinstance(output_df.dtypes[col], TensorDtype):
             output_df[col] = output_df[col].to_numpy()
+        # # DL predictor outputs raw ndarray outputs as opaque numpy object.
+        # # ex: output_df = pd.DataFrame({"predictions": [np.array(1)]})
+        elif output_df.dtypes[col] == np.dtype(object):
+            output_df[col] = np.array([np.asarray(v) for v in output_df[col]]).tolist()
```

Comment:
do we want to do this for _all_ objects? this should only be done for numpy arrays right?

Reply:
Like a string value has object dtype.

Reply:
+1, I'm assuming that we only want to do this if all elements are ndarrays.

Reply:
sg we can enhance assertion here a bit. 

We should really have another PR to address air_integrations.py adapters after all numpy preprocessor + predictor pieces are merged with clear story of TensorArray now. This file is pre Ray 2.0 and I don't think anyone is actively maintaining it ..